econml/metalearners/_metalearners.py

```diff
@@ -133,7 +147,10 @@ def __init__(self, *,
                  overall_model,
                  categories='auto'):
         self.overall_model = clone(overall_model, safe=False)
+        if categories != 'auto':
+            categories = [categories]  # OneHotEncoder expects a 2D array with features per column
         self.categories = categories
+        self.transformer = OneHotEncoder(categories=self.categories, sparse=False, drop='first')
```

Comment:
Good point. The fundamental problem here is that `categories` is set during `__init__` or at some other point by the user rather than during `fit` despite the fact that `categories` is a property of `T`. I fixed this by implementing a special setter for `categories` in `TreatmentExpansionMixin`, see latest commit.